cube/src/main/java/zone/cogni/asquare/cube/convertor/data2shacl/ShaclGenerator.java

```diff
@@ -172,15 +172,21 @@ private Resource calculateShapeBasedOnResource(@Nonnull Configuration configurat
 
     String namespacePrefix = shacl.getNsURIPrefix(originalResource.getNameSpace());
     if (namespacePrefix == null) {
-      throw new RuntimeException("no name alternative found for '" + originalResource.getURI() + "':"
-                                 + " please add namespace to prefixes.");
+      namespacePrefix = "ns" + getAvailableNamespaceIndex(shacl);
+      shacl.setNsPrefix(namespacePrefix, originalResource.getNameSpace());
     }
 
     String prefixLocalName = firstPart == null ? namespacePrefix + "_" + localName
                                                : firstPart + "_" + namespacePrefix + "_" + localName;
     return ResourceFactory.createResource(configuration.getShapesNamespace() + prefixLocalName);
   }
 
+  private long getAvailableNamespaceIndex(Model shacl) {
+    return shacl.getNsPrefixMap().keySet().stream()
+            .filter(key-> key.matches("^ns[0123456789]*$"))
+            .count();
```

Comment:
Unfortunately, count is not a good way of calculating things. what if you have only ns2? @michielrogissart-cognizone